utility/unit_of_work/module.go

```diff
@@ -216,3 +113,46 @@ func (uow *baseUtilityUnitOfWork) Release() error {
 
 	return nil
 }
+
+func (uow *baseUtilityUnitOfWork) isProposalBlockSet() bool {
+	return uow.proposalStateHash != "" && uow.proposalProposerAddr != nil
+}
+
+// processTransactionsFromProposalBlock processes the transactions from the proposal block.
+// It also removes the transactions from the mempool if they are already present.
+func (uow *baseUtilityUnitOfWork) processTransactionsFromProposalBlock(txMempool mempool.TXMempool, txsBytes [][]byte) (err error) {
```

Comment:
```suggestion
func (uow *baseUtilityUnitOfWork) processTransactionsFromProposalBlock(txMempool mempool.TXMempool) error {
```

Get the txs from the `uow` to avoid discrepancies